libvncserver/ws_decode.h

```diff
@@ -0,0 +1,151 @@
+#ifndef _WS_DECODE_H_
```

Comment:
One of the challenges with using an existing websockets library is that the current code automatically detects whether the connection is regular VNC, websockets unencrypted, websockets encrypted, or flash policy request (this last can of course go away if the Flash fallback client is dropped). This is somewhat unusual for servers to do this and it is definitely complicated by the fact that in RFB the server speaks first (which is not  considered good practice for many reasons). This functionality is often difficult to shoehorn into existing libraries (sniffing initial bytes on a connection without pulling them off the stream is a cross-platform challenge). I agree that having a custom implementation of a websockets server in libvncserver does increase the security surface area. On the other hand, as someone who also works in the ops/deployment space, reducing the ports that have to be managed/firewalled is a big advantage that shouldn't be discounted. Also, removing Hixie will significantly decrease the attack surface. One of the huge issues with Hixie was that you couldn't allocate buffers ahead of time by just inspecting a few bytes of the header. So only having to support HyBi reduces many potential buffer management problems. Anyways, just some thoughts.